Common/TradingCalendar.cs

```diff
@@ -0,0 +1,133 @@
+            var today = _securityManager.UtcTime.Date;
+
+            return GetTradingDay(today);
+        }
+
+        /// <summary>
+        /// Method returns <see cref="TradingDay"/> that contains trading events associated with the given date
+        /// </summary>
+        /// <returns>Populated instance of <see cref="TradingDay"/></returns>
+        public TradingDay GetTradingDay(DateTime day)
+        {
+            return GetTradingDays(day, day).First();
+        }
+
+        /// <summary>
+        /// Method returns <see cref="TradingDay"/> that contains trading events associated with the range of dates
+        /// </summary>
+        /// <param name="start">Start date of the range (inclusive)</param>
+        /// <param name="end">End date of the range (inclusive)</param>
+        /// <returns>>Populated list of <see cref="TradingDay"/></returns>
+        public IEnumerable<TradingDay> GetTradingDays(DateTime start, DateTime end)
+        {
+            return PopulateTradingDays(start, end);
+        }
+
+        /// <summary>
+        /// Method returns <see cref="TradingDay"/> of the specified type (<see cref="TradingDayType"/>) that contains trading events associated with the range of dates
+        /// </summary>
+        /// <param name="type">Type of the events</param>
+        /// <param name="start">Start date of the range (inclusive)</param>
+        /// <param name="end">End date of the range (inclusive)</param>
+        /// <returns>>Populated list of <see cref="TradingDay"/></returns>
+        public IEnumerable<TradingDay> GetDaysByType(TradingDayType type, DateTime start, DateTime end)
+        {
+            Func<TradingDay, bool> typeFilter = day =>
+                {
+                    switch (type)
+                    {
+                        case TradingDayType.BusinessDay:
+                            return day.BusinessDay;
+                        case TradingDayType.PublicHoliday:
+                            return day.PublicHoliday;
+                        case TradingDayType.Weekend:
+                            return day.Weekend;
+                        case TradingDayType.OptionExpiration:
+                            return day.OptionExpirations.Any();
+                        case TradingDayType.FutureExpiration:
+                            return day.FutureExpirations.Any();
+                        case TradingDayType.FutureRoll:
+                            return day.FutureRolls.Any();
+                        case TradingDayType.SymbolDelisting:
+                            return day.SymbolDelistings.Any();
+                        case TradingDayType.EquityDividends:
+                            return day.EquityDividends.Any();
+                    };
+                    return false;
+                };
+            return GetTradingDays(start, end).Where(typeFilter);
+        }
+
+
+        private IEnumerable<TradingDay> PopulateTradingDays(DateTime start, DateTime end)
```

Comment:
I assume these are in utc? We should always make this explicit since time zones can make a happy day very sad, lol.

Also, in the loop below, should we be converting to the symbol's time zone before defining the trading day?  UTC date is not the same as an EDT date